cli/pkg/local/app.go

```diff
@@ -182,9 +184,12 @@ func NewApp(ctx context.Context, opts *AppOptions) (*App, error) {
 		"rill.download_row_limit": "0", // 0 means unlimited
 		"rill.stage_changes":      "false",
 	}
-	for k, v := range opts.Variables {
-		vars[k] = v
+	maps.Copy(vars, opts.Variables)
+	envMap, err := godotenv.Read(filepath.Join(projectPath, ".env"))
+	if err != nil && !errors.Is(err, os.ErrNotExist) {
+		return nil, err
 	}
+	maps.Copy(vars, envMap)
```

Comment:
**Performance Issue**: The use of godotenv.Read to load environment variables from a file during runtime can lead to performance issues due to I/O operations. This is especially critical in a function that might be called frequently or in a performance-sensitive context. <br> **Fix**: To enhance performance, consider caching the environment variables at application startup and using in-memory storage for subsequent accesses. This avoids repeated I/O operations. <br> **Code Suggestion**: 
 ```
 -	envMap, err := godotenv.Read(filepath.Join(projectPath, ".env"))
 -	if err != nil and !errors.Is(err, os.ErrNotExist) {
 -		return nil, err
 -	}
 -	maps.Copy(vars, envMap)
 +	if envMap == nil {
 +		envMap, err = godotenv.Read(filepath.Join(projectPath, ".env"))
 +		if err != nil and !errors.Is(err, os.ErrNotExist) {
 +			return nil, err
 +		}
 +	}
 +	maps.Copy(vars, envMap)
 ```